gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/engine/VertxPluginRegistry.java

```diff
@@ -72,30 +93,73 @@ public VertxPluginRegistry(Vertx vertx, VertxEngineConfig vxEngineConfig, Map<St
      */
     @Override
     protected void downloadArtifactTo(final URL artifactUrl, final File pluginFile,
-            final IAsyncResultHandler<File> handler) {
+                                      final IAsyncResultHandler<File> handler) {
+
+        HttpClient client;
+        HttpClientOptions clientOpts=new HttpClientOptions();
+
         int port = artifactUrl.getPort();
-        if (port == -1) {
-            port = 80;
+
+        //Configure http client options following artifact url
+        if (artifactUrl.getProtocol().equals("https")) {
+            //Enable SSL
+            clientOpts.setSsl(true);
+
+            //If port is not defined, set to https default port 443
+            if (port == -1) port = 443;
+
+            //If artifact host is localhost
+            if(artifactUrl.getHost().equals("localhost") ||artifactUrl.getHost().equals("127.0.0.1")){
```

Comment:
Style nit: spaces on if statement, including `||`

Generally spaces around all these constructs (I'll stop pointing each one out, I'm sure you get the idea!).

Reply:
I'm slightly concerned this test for just localhost isn't robust enough and we should be using a better mechanism that will support loopback in general (i.e. not just those specific addresses, and also IPv6).

I know there's InetAddress#isAnyLocalAddress and InetAddress#isLoopbackAddress. Thoughts?

Reply:
my apologies for the style. I will correct the spaces.

About the test for localhost, you're right. We must consider the local loopback and local addresses, whatever the expression. I will test `netAddress#isAnyLocalAddress` and `InetAddress#isLoopbackAddress`.

Reply:
`InetAddress#isLoopbackAddress` is detected well loopback by ip or hostname.

But `netAddress#isAnyLocalAddress` is not detecting local machine addresses. It report if ip is wildcard, as 0.0.0.0.

So, to check if artifact url hostname is local address, I build a list of local ip addresses, for comparison.